sources/nvidia-migmanager/src/main.rs

```diff
@@ -0,0 +1,688 @@
+/*!
+# NVIDIA MIG Manager
+`nvidia-migmanager` ensures that MIG settings are applied to an instance that supports
+it. It is called by `nvidia-migmanager.service`.
+
+The binary first checks if MIG is activated by checking the value of
+`settings.kubelet-device-plugins.nvidia.device-partitioning-strategy`. Then, it activates
+MIG and applies the profile according to the type of GPU present in the instance.
+
+NVIDIA MIG is currently supported only in A30, A100, H100 and H200 GPUs.
+
+## Example:
+```toml
+[settings.kubelet-device-plugins.nvidia]
+device-partitioning-strategy="mig"
+
+[settings.kubelet-device-plugins.nvidia.mig.profile]
+"a100.40gb"="2"
+"h100.80gb"="4"
+"h200.141gb"="3"
+```
+This would partition the GPUs in an instance with A100 GPU into 2 parts, instance with H100
+into 4 parts and instance with H200 into 3 parts.
+*/
+
+use argh::FromArgs;
+use log::{info, trace, warn};
+use regex::Regex;
+use serde::Deserialize;
+use simplelog::{Config as LogConfig, LevelFilter, SimpleLogger};
+use snafu::{ensure, ResultExt};
+use std::cmp::min;
+use std::collections::HashMap;
+use std::ffi::OsStr;
+use std::fs;
+use std::path::Path;
+use std::process::{self, Command};
+
+use bottlerocket_modeled_types::{MigProfile, NvidiaDevicePartitioningStrategy, NvidiaGpuModel};
```

Comment:
I'm not sure we benefit enough from these imports to take on the dependency.

`MigProfile` gets turned immediately back into a string and then turned into one of the specialized types here, so we could just keep it as a string.

`NvidiaDevicePartitioningStrategy` - we only use the `MIG` variant, so we could just check for that string value.

`NvidiaGpuModel` could probably be kept as a string and turned into `NvidiaGpu` when needed.